### ISSUE TITLE
[BUG] Prometheus Monitoring is showing "Page not found 404 Error" in cluster home

### ISSUE THREAD

**Issue by eroji** [rancher/rancher#41562]
**Rancher Server Setup**
- Rancher version: 2.7.1
- Installation option (Docker install/Helm Chart): Helm
   - If Helm Chart, Kubernetes Cluster and version (RKE1, RKE2, k3s, EKS, etc): RKE1
- Proxy/Cert Details:

**Information about the Cluster**
- Kubernetes version: 1.20.15
- Cluster Type (Local/Downstream): Downstream
   - If downstream, what type of cluster? (Custom/Imported or specify provider for Hosted/Infrastructure Provider): Custom

**User Information**
- What is the role of the user logged in? (Admin/Cluster Owner/Cluster Member/Project Owner/Project Member/Custom)
  - If custom, define the set of permissions: Admin

**Describe the bug**

Cluster home with Monitoring installed is showing "Page not found 404 Error".

**To Reproduce**

Installed Prometheus Monitoring app from Apps > Charts > Monitoring. After everything is installed, navigate to the Cluster home. Scroll to the bottom and the Cluster Metrics, Kubernetes Component Metrics and Etcd Metrics should now be displayed. However, it is showing "Page not found 404 Error".

**Result**
No Grafana graphs are shown, only "Page not found 404 Error".

**Expected Result**

The respective Grafana graphs should be displayed

**Screenshots**

![image](https://user-images.githubusercontent.com/13228540/230748850-d6be97b7-ed42-4692-9142-35f843e717a3.png)

**Additional context**

**Comment by nflynt** [comment]
Did some digging on this with @mantis-toboggan-md and figured out that the requirement for the `/k8s/clusters/local` prefix has changed between Monitoring chart versions. This was related to some troubleshooting around the underlying proxy and nginx configuration, which has since been fixed.

In short:
For specific chart versions `100.2.0+up40.1.2` and `102.0.0+up40.1.2`, links to Grafana should include the `/k8s/clusters/local` prefix
For any other chart version, including newer versions, links to Grafana should **not** include the `/k8s/clusters/local` prefix.

Both the link on Monitoring -> Grafana, and the embedded iframe views on the Cluster tab, should follow the same logic regarding the prefix. Right now these two locations appear to be using different rules to construct the URL, usually resulting in one or the other being broken for a given chart version.

**Comment by maxgio92** [comment]
Hello everyone, sharing my experience, it happens when for the "rancher-monitoring" App/Helm release (which is for 99% the [kube-prometheus-stack](https://github.com/prometheus-community/helm-charts/tree/main/charts/kube-prometheus-stack)), the Helm values:

- `global.cattle.clusterId`
- `global.cattle.clusterName`

are not set.

Those values are needed to build the Nginx rewrite rules correctly, to proxy the Grafana Kubernetes Service in the downstream clusters through the downstream cluster's API server proxy via the Rancher proxy.

**Comment by AntoineNGUYEN31** [comment]
@maxgio92  you are right. Update the helm option & redepoy all pods works like a charm.
For those you use fleet.yaml to deploy here is the value to put:
global:
  cattle:
    clusterId: global.fleet.clusterLabels.management.cattle.io/cluster-name
    clusterName: global.fleet.clusterLabels.management.cattle.io/cluster-display-name
